SwiftCompilerSources/Sources/AST/DiagnosticEngine.swift

```diff
@@ -0,0 +1,117 @@
+//===--- DiagnosticEngine.swift -------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2022 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+@_exported import ASTBridging
+import Basic
+
+public typealias DiagID = BridgedDiagID
+
+extension BridgedDiagnosticArgument {
+  init(stringRef val: BridgedStringRef) {
+    self.init(kind: .stringRef, value: .init(stringRefValue: val))
+  }
+  init(int val: Int) {
+    self.init(kind: .int, value: .init(intValue: val))
+  }
+  init(uInt val: UInt) {
```

Comment:
Do we really need UInt?

Reply:
Not entirely sure.
Even if the C++ diagnostic ID declared to receive an `uint` argument, as far as I can read from the DiagnosticEngine C++ code, it accepts `DiagnosticArgument(int)` value too. So maybe it's not needed. Since we can add it later, I just removed it for now.